src/stackset-stack.ts

```diff
@@ -12,21 +12,94 @@ import {
   Names,
   Lazy,
   FileAssetPackaging,
+  App,
+  Resource,
+  Annotations,
 } from 'aws-cdk-lib';
+import { CfnBucket, IBucket } from 'aws-cdk-lib/aws-s3';
+import { BucketDeployment, Source } from 'aws-cdk-lib/aws-s3-deployment';
 import { Construct } from 'constructs';
 
+export const fileAssetResourceName = 'StackSetAssetsBucketDeployment';
```

Comment:
this is used needed so we can export the construct name so we can find it after creating our StackSet resource so we can add a dependency. the bucket deployment must finish before the StackSet resource is created. Open to ideas on better naming conventions and/or better ways to accomplish this.

Reply:
We could make the `bucketDeployment` property public and then we could access
it.

```ts
(Stack.of(this).synthesizer as StackSetStackSynthesizer).bucketDeployment
```